src/model/kaniCommandCreate.ts

```diff
@@ -98,26 +74,14 @@ export async function captureFailedChecks(
 	args?: number,
 ): Promise<KaniResponse> {
 	let harnessCommand = '';
-	if (args === undefined) {
-		harnessCommand = `${KaniConstants.KaniExecutableName} ${rsFile} ${KaniArguments.harnessFlag} ${harnessName}`;
-	} else {
-		harnessCommand = `${KaniConstants.KaniExecutableName} ${rsFile} ${KaniArguments.harnessFlag} ${harnessName} ${KaniArguments.unwindFlag} ${args}`;
-	}
-	const kaniOutput = await createFailedDiffMessage(harnessCommand);
-	if (kaniOutput.failedProperty == 'error') {
-		const crateURI = getRootDir();
-		let harnessCommand = '';
 
-		if (args === undefined || NaN) {
-			harnessCommand = `${KaniConstants.CargoKaniExecutableName} ${KaniArguments.harnessFlag} ${harnessName} --output-format terse`;
-		} else {
-			harnessCommand = `${KaniConstants.CargoKaniExecutableName} ${KaniArguments.harnessFlag} ${harnessName} ${KaniArguments.unwindFlag} ${args} --output-format terse`;
-		}
-		const kaniOutput = await createFailedDiffMessage(harnessCommand);
-		return kaniOutput;
+	if (args === undefined || NaN) {
```

Comment:
Does this condition need to be `args === undefined || args === NaN`?

Reply:
Yes, since args is an optional number (although this will be modified by the tree-sitter anyway), which means it can be a number or undefined. But even if it's passing a number, typescript encodes errors as numbers as a NaN value (not a number). So we are checking for cases where either there is no args or there is an args but there is an incorrect or erroneous value.